go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -92,6 +89,25 @@ func getLbEndpoints(slices []*discovery.EndpointSlice, svcPort v1.ServicePort, f
 }
 
 func deleteVIPsFromOVN(vips sets.String, st *serviceTracker, name, namespace, clusterPortGroupUUID string) error {
+	deleteVIPsFromOVNBalancers(vips, st, name, namespace)
```

Comment:
I can't remember when we did this but we should not inject the service tracker as dependency, those operations should happen in the main loop

Reply:
So (apart from removing the dependency from where not used) you are suggesting to do the loop explicitly in the reconciliation function right?
I agree better not burning it in a function.

Reply:
yeah, in this case it seems more readable in the main loop, and it really doesn't belong to `deleteVIPsFromOVN` , semantically I don't expect this function to remove the things from the tracker

Reply:
+1